components/sup/src/manager/mod.rs

```diff
@@ -1741,6 +1780,30 @@ impl Manager {
     }
 }
 
+fn tls_config<P>(key_path: P, cert_path: P) -> Result<ServerConfig>
+where
+    P: AsRef<Path>,
```

Comment:
This is fine, and doesn't need to change, but for future reference, this means that both `key_path` and `cert_path` will have to resolve to the same concrete type (i.e., both are `str`, both are `String`, both are `PathBuf`, etc.). If one were a `String` but the other were a `PathBuf`, say, it wouldn't compile, because `P` couldn't be both of those things simultaneously.

To get around that, you'd just declare two different trait bounds, one for each argument you're matching, e.g.
```rust
...
where
    K: AsRef<Path>,
    C: AsRef<Path>,
...
```
and then the compiler could cope with all the potential combinations.

Reply:
Ah, of course.  It's fine, I can update it.